data/scripts.js

```diff
@@ -900,6 +900,127 @@ exports.BattleScripts = {
 
 		return team;
 	},
+	queryMoves: function (moves, hasType, hasAbility) {
+		// This is primarily a helper function for random setbuilder functions.
+		var counter = {
+			Physical: 0, Special: 0, Status: 0, damage: 0, recovery: 0, stab: 0,
+			blaze: 0, overgrow: 0, swarm: 0, torrent: 0,
+			adaptability: 0, ate: 0, bite: 0, contrary: 0, hustle: 0,
+			ironfist: 0, serenegrace: 0, sheerforce: 0, skilllink: 0, technician: 0,
+			inaccurate: 0, priority: 0, recoil: 0,
+			physicalsetup: 0, specialsetup: 0, mixedsetup: 0, speedsetup: 0,
+			damagingMoves: [],
+			damagingMoveIndex: {},
+			setupType: ''
+		};
+
+		if (!moves || !moves.length) return counter;
+		if (hasType === undefined) hasType = false;
+		if (hasAbility === undefined) hasAbility = false;
```

Comment:
You recommended `false` for the `affectedByImmunities` refactor, since that would be run quite a few times and initialize quite a few objects all the time (and the code checked for that anyhow).

For this, it should be fine to initialize the object as a just-in-case crashguard and really it should never be reached anyways.